File: cpp/RT_CPP/RT_CPP/image_generator.cpp
#include "image_generator.h"
#include "io/EasyBMP.cpp"

using namespace utility::io;

void image_generator::write_pixels(vector3** pixels, int width, int height, string output)
{
	BMP input;
	input.SetSize(width, height);
	for (int y = 0; y < height; y++)
	{
		for (int x = 0; x < width; x++)
		{
			
			input(x, y)->Alpha = 255;
			input(x, y)->Red = static_cast<unsigned char>(static_cast<int>(pixels[y][x].x * 255));
			input(x, y)->Green = static_cast<unsigned char>(static_cast<int>(pixels[y][x].y * 255));
			input(x, y)->Blue = static_cast<unsigned char>(static_cast<int>(pixels[y][x].z * 255));

		}
	}
	input.WriteToFile(output.c_str());
}

File: Csharp/RT_CS/RT_CS/utility/IO/ImageGenerator.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Drawing.Imaging;
using System.Text;

namespace RT_CS.utility.IO
{
    public class ImageGenerator
    {
        public static void WritePixels(Vector3[,] pixels, int width, int height, string output)
        {
            using Bitmap image = new Bitmap(width, height);
            for (int y = 0; y < height; y++)
            {
                for (int x = 0; x < width; x++)
                {
                    image.SetPixel(x, y, VecColor(pixels[y, x]));
                }
            }
            image.Save(output, ImageFormat.Bmp);
        }

        private static Color VecColor(Vector3 pixel)
        {
            return Color.FromArgb(
                (int)(255 * pixel.x), (int)(255 * pixel.y), (int)(255 * pixel.z)
                );
        }
    }
}
